### ISSUE TITLE
[perf] jitted_run_model cpp fastpath miss every decode step under dp=1 + Pathways proxy backend

### ISSUE THREAD

**Issue by zorrofox** [sgl-project/sglang-jax#1452]
# [perf] `jitted_run_model` cpp fastpath miss every decode step under dp=1 + Pathways proxy backend

## Symptom

On GLM-5.2-FP8 (epmoe/MLA) with `--dp-size=1` under Pathways single-controller (`JAX_PLATFORMS=proxy`), every warm decode step reports `#cache_miss: 1` from `count_pjit_cpp_cache_miss()` around `jitted_run_model`. Observed on both **colocated** and PD-disaggregated configs — this is not PD-specific.

Not observed with `--dp-size=2` (MiMo-V2 tp8 dp2: `#cache_miss: 0` in steady state).

Overhead: ~<1ms/step (<1.1% of 89ms ITL on GLM tp16), so not a throughput blocker, but it defeats the pjit cpp fastpath and forces the Python slow path on every call.

## Root cause

`ForwardBatch.from_model_worker_batch` (via `device_array`) constructs a **fresh `NamedSharding` object per call** for each metadata array (`input_ids`, `positions`, `seq_lens`, `cache_loc`, ...). Instrumented `id(sharding)` differs every step; `id(sharding.mesh)` is stable.

jaxlib `PjitFunctionCache` key (`CallSignature`, `jaxlib/jax_jit.{h,cc}` @ jax-v0.8.1):
- `HashShardingForJit(NamedSharding)` = `absl::Hash<void*>()(mesh().ptr())` — stable ✓
- `EqualShardingsForJit(NamedSharding a, b)`:
  - fast-path `a.ptr() == b.ptr()` — **fails** (fresh object each call)
  - slow-path compares `mesh().ptr()` (stable) + `spec` (stable) + `memory_kind.equal()` + `logical_device_ids().equal()`

One of `memory_kind` / `logical_device_ids` / `layout` (also in `CallSignature`) differs across fresh `NamedSharding` objects under dp=1 (size-1 `data` axis). Under dp=2 the same fields are stable, so the slow-path `==` returns true and the cache hits.

Verified: `_get_fastpath_data` returns non-None (so it's not an output-type or effects issue); `JAX_EXPLAIN_CACHE_MISSES` shows tracing-cache hits after the first per-bs call — the miss is purely the cpp `PjitFunctionCache` lookup.

## Repro

```bash
python -m sgl_jax.launch_server \
  --model-path <GLM-5.2-FP8> --tp-size 16 --ep-size 16 --dp-size 1 \
  --page-size 128 --attention-backend fa --moe-backend epmoe \
  --disable-radix-cache --mem-fraction-static 0.86 \
  --precompile-bs-paddings 1 4 8 16 32 --precompile-token-paddings 4096
# any load; every "Decode batch" log line shows "#cache_miss: 1"
```

## Proposed fix

Cache and reuse `NamedSharding` objects in `device_array` (or at the `ForwardBatch`/`ModelWorkerBatch` level) keyed on `(mesh, spec)`, so `sharding.ptr()` is stable across calls and `EqualShardingsForJit`'s fast-path `a.ptr() == b.ptr()` short-circuits true. This also avoids the per-call `NamedSharding.__init__` overhead.

Alternative: pin `in_shardings` on `jitted_run_model` so jax reshards inputs to a fixed sharding object before the cache lookup.

## Environment

- jax 0.8.1, Pathways proxy backend (`JAX_PLATFORMS=proxy`)
- TPU multi-host, 1 slice tp16 dp1
- sglang-jax main @ 0381ccbf
